torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -481,12 +481,41 @@ def _precompile_config(self, cfg: Config, warm_cache_only: bool):
                 raise
             binary._init_handles()
 
+        """
+        https://github.com/pytorch/pytorch/issues/115344
+
+        self.fn.constexprs doesn't properly deal with None args, so when we filter out
+        an arg in UserDefinedTritonKernel.codegen, we need to filter it here as well.
+        We also don't want to modify self.fn.
+
+        We know that we removed something from the signature if:
+            1. It's in compile_meta["constants"]
+            2. It isn't a constant we already know about
+                Note: The value of interest has already been added to compile_meta['constants'],
+                    so we use self.fn.constexprs instead.
+            3. It isn't in the compile_meta signature
+        """
+        self.arg_names = self.fn.arg_names
+        removed_from_sig = set(compile_meta["constants"].keys())
+        known_constants = {
+            arg for i, arg in enumerate(self.arg_names) if i in self.fn.constexprs
+        }
+        removed_from_sig = removed_from_sig.difference(known_constants)
+        removed_from_sig = removed_from_sig.difference(
+            set(compile_meta["signature"].keys())
```

Comment:
Can `compile_meta["signature"].keys()` intersect with `compile_meta["constants"].keys()` (used as the base for `removed_from_sig`? From the code in the `wrapper.py` seems disjoint.

Reply:
I think I address this here: https://github.com/pytorch/pytorch/pull/138260#discussion_r1806760766